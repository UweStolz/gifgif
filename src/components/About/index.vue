<template>
  <v-container fluid>
    <v-row justify="center">
      <v-card
        tile
        flat
        :width="$vuetify.breakpoint.xsOnly ? '90%' : '50%'"
      >
        <v-card-title class="display-2 justify-center">
          <router-link
            style="color: black;"
            to="/"
          >
            GIFGIF
          </router-link>
        </v-card-title>
        <vue-markdown :source="cardText" />
      </v-card>
    </v-row>
  </v-container>
</template>

<script lang="ts">
import { Vue, Component } from 'vue-property-decorator';
// @ts-ignore - No types available
import VueMarkdown from 'vue-markdown';

// eslint-disable-next-line @typescript-eslint/no-var-requires
const vueLogo = require('@/assets/vue.svg');
// eslint-disable-next-line @typescript-eslint/no-var-requires
const vuetifyLogo = require('@/assets/vuetify.svg');
// eslint-disable-next-line @typescript-eslint/no-var-requires
const githubLogo = require('@/assets/github.svg');

@Component({
  components: {
    'vue-markdown': VueMarkdown,
  },
})
export default class About extends Vue {
  cardText = `
  ---
  
  ## Privacy Policy
  No Personal Data will be Collected!

  ### Browser Storage
  Application information will be saved in the IndexedDB of your Browser, via the [IndexedDB API.](https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API)
  This information is **only** available to **you**.

  ## Copyright
  This project is licensed under the [MIT license](https://github.com/UweStolz/gifgif/blob/master/LICENSE.md)

  ## References

  ### Technologies
  This project was build with:  
  <table>
  <tr>
  <td><a href="https://vuejs.org/"><img src="${vueLogo}" width="64"></a></td>
  <td>Vue</td>
  </tr>
  <tr>
  <td><a href="https://vuetifyjs.com/"><img src="${vuetifyLogo}" width="64"></a></td>
  <td>Vuetify</td>
  </tr>
  </table>

  ### APIs
  [Giphy](https://developers.giphy.com/docs/api/)
  [Tenor](https://tenor.com/gifapi/documentation)

  ### Repository:
  <a href="https://github.com/UweStolz/gifgif"><img src="${githubLogo}" width="64"></a> Available under GitHub
  `;
}
</script>
