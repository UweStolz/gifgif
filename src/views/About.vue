<template>
  <AboutView />
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';
import AboutView from '../components/About/index.vue';

@Component({
  components: {
    AboutView,
  },
})
export default class About extends Vue { }
</script>
