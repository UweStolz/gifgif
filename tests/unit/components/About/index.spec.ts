import About from '@/components/About/index.vue';
import shallow from '../../../helper';

describe('About.vue', () => {
  it('renders properly', () => {
    const wrapper = shallow(About);
    expect(wrapper.element).toMatchSnapshot();
  });
});
