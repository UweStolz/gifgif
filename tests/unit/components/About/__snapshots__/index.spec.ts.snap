// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About.vue renders properly 1`] = `
<v-container-stub
  fluid="true"
  tag="div"
>
  <v-row-stub
    justify="center"
    tag="div"
  >
    <v-card-stub
      flat="true"
      loaderheight="4"
      tag="div"
      tile="true"
      width="50%"
    >
      <v-card-title-stub
        class="display-2 justify-center"
      >
        <router-link
          style="color: black;"
          to="/"
        >
          
          GIFGIF
        
        </router-link>
      </v-card-title-stub>
       
      <vue-markdown-stub
        anchorattributes="[object Object]"
        breaks="true"
        emoji="true"
        highlight="true"
        html="true"
        langprefix="language-"
        linkify="true"
        postrender="function _default(htmlData) {
	        return htmlData;
	      }"
        prerender="function _default(sourceData) {
	        return sourceData;
	      }"
        quotes="“”‘’"
        show="true"
        source="
  ---
  
  ## Privacy Policy
  No Personal Data will be Collected!

  ### Browser Storage
  Application information will be saved in the IndexedDB of your Browser, via the [IndexedDB API.](https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API)
  This information is **only** available to **you**.

  ## Copyright
  This project is licensed under the [MIT license](https://github.com/UweStolz/gifgif/blob/master/LICENSE.md)

  ## References

  ### Technologies
  This project was build with:  
  <table>
  <tr>
  <td><a href=\\"https://vuejs.org/\\"><img src=\\"\\" width=\\"64\\"></a></td>
  <td>Vue</td>
  </tr>
  <tr>
  <td><a href=\\"https://vuetifyjs.com/\\"><img src=\\"\\" width=\\"64\\"></a></td>
  <td>Vuetify</td>
  </tr>
  </table>

  ### APIs
  [Giphy](https://developers.giphy.com/docs/api/)
  [Tenor](https://tenor.com/gifapi/documentation)

  ### Repository:
  <a href=\\"https://github.com/UweStolz/gifgif\\"><img src=\\"\\" width=\\"64\\"></a> Available under GitHub
  "
        tableclass="table"
        tasklists="true"
        tocanchorclass="toc-anchor"
        tocanchorlink="true"
        tocanchorlinkclass="toc-anchor-link"
        tocanchorlinkspace="true"
        tocanchorlinksymbol="#"
        tocclass="table-of-contents"
        tocfirstlevel="2"
        typographer="true"
        watches="source,show,toc"
        xhtmlout="true"
      />
    </v-card-stub>
  </v-row-stub>
</v-container-stub>
`;
